src/tpm2_pytss/ESAPI.py

```diff
@@ -6867,6 +6924,39 @@ def tr_deserialize(self, buffer: bytes) -> ESYS_TR:
 
         return ESYS_TR(esys_handle[0])
 
+    def _incr_handle_ref(self, handle):
+        if handle not in self._ref_handles:
+            return
+        tracked, count = self._ref_handles[handle]
+        count += 1
+        self._ref_handles[handle] = (tracked, count)
+
+    def _decr_handle_ref(self, handle):
+        if handle not in self._ref_handles:
+            return
+        tracked, count = self._ref_handles[handle]
+        count -= 1
+        if count <= 0 and tracked:
+            self.flush_context(handle)
+
+    def _set_tracking_ref(self, handle, tracked):
+        if handle not in self._ref_handles:
+            return
+        _, count = self._ref_handles[handle]
+        self._ref_handles[handle] = (tracked, count)
+
+    def _handle_with_ref(self, handle: ESYS_TR) -> ESYS_TR:
+        """Set up weak references between ESAPI context and handle.
+
+           If ectx._flush_on_close is True create weak references so we can flush on close/unref.
+        """
+        if not self._flush_on_close:
+            return handle
+        handle._ectx_ref = weakref.ref(self)
+        self._ref_handles[handle] = (True, 0)
```

Comment:
could you comment on what this tuple structure represents for clarity while reading the code.

Reply:
I'll add inline comments later on, but for now:
The tuple is (tracked, reference_count), the tracked value is used to check if a session has TPMA_SESSION.CONTINUESESSION set or not, and as it can be toggled at any time the reference counter needs to be saved